Kernel/Arch/x86/Processor.h

```diff
@@ -92,6 +92,7 @@ class Processor {
     bool m_invoke_scheduler_async;
     bool m_scheduler_initialized;
     bool m_in_scheduler;
+    bool m_allocation_enabled;
```

Comment:
I realized just now that my suggestion was dumb. This flag should be in the `Thread` class instead.